src/dsll_progs/algebra.dsll

```diff
@@ -0,0 +1,15 @@
+constructor Vector : [VectorSpace] -> type                    
+constructor Scalar : type                                               
+constructor Map : [type, type] -> type
+constructor VectorSpace : type                                      
+
+operator AddV : forvars [V] : [VectorSpace], [Vector(V), Vector(V)] -> Vector(V)
+operator AddS : [Scalar, Scalar] -> Scalar
+operator Norm : forvars [V] : [VectorSpace], [Vector(V)] -> Scalar
+operator InnerProd : forvars [V] : [VectorSpace], [Vector(V), Vector(V)] -> Scalar
+operator Apply : fortypes [A,B] : [type,type], [Map(A, B), A] -> B
```

Comment:
There's an inconsistency between line 10 and line 12 (due to the inconsistent way I originally wrote the book):
`fortypes [A,B] : [type,type]` vs. `forvars [U,V] : [VectorSpace]`
I'd say make all types explicit for now (don't allow multiple declaration). We can add concision later.
Also, if there's a long list of quantifiers, it might be better to change the syntax to
`forvars [(v : T)]` and `fortypes [(X : Type)]` (i.e. each variable is given its type, rather than having two separate lists). I think that's more readable.

Reply:
Yes, the syntax change you suggested is much more readable